File: home-server-ng/src/main/java/maroroma/homeserverng/kiosk/model/weather/AllForeCasts.java
package maroroma.homeserverng.kiosk.model.weather;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.*;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class AllForeCasts {
    private CurrentForecast currentForecast;
    private List<Forecast> forecastsForTheCurrentDay;
    private List<DailyForecast> futureForeCasts;
}

File: home-server-ng/src/frontend/home-server-tsx/src/model/kiosk/AllForeCasts.ts
import CurrentForecast from "./CurrentForecast";
import DailyForecast from "./DailyForecast";
import Forecast from "./Forecast";

export default class AllForeCasts {

    static empty(): AllForeCasts {
        return new AllForeCasts(CurrentForecast.empty(), [], []);
    }

    constructor(
        public currentForecast: CurrentForecast,
        public forecastsForTheCurrentDay: Forecast[],
        public futureForeCasts: DailyForecast[]
    ) { }
}